server/http.go

```diff
@@ -411,35 +683,63 @@ func (p *Plugin) handleStartMeeting(w http.ResponseWriter, r *http.Request) {
 		return
 	}
 
-	client, _, err := p.getActiveClient(user)
-	if err != nil {
-		p.API.LogWarn("Error getting the client", "err", err)
-		return
+	// topic
+	topic := req.Topic
+	if topic == "" {
+		topic = defaultMeetingTopic
 	}
 
-	meeting, err := client.CreateMeeting(zoomUser, defaultMeetingTopic)
-	if err != nil {
-		p.API.LogWarn("Error creating the meeting", "err", err)
-		return
+	usePMI := req.UsePMI
+	var meetingID int = -1
+	var createMeetingErr error = nil
+	switch usePMI {
+	case "":
+		if userPMISettingPref, getUserPMISettingErr := p.getPMISettingData(user.Id); getUserPMISettingErr == nil {
+			switch userPMISettingPref {
+			case zoomPMISettingValueAsk:
+				p.askUserPMIMeeting(user.Id, req.ChannelID)
+			case "", trueString:
+				meetingID = zoomUser.Pmi
+			default:
+				meetingID, createMeetingErr = p.createMeetingWithoutPMI(user, zoomUser, req.ChannelID, topic)
+			}
+		} else {
+			p.askUserPMIMeeting(user.Id, req.ChannelID)
+		}
+	case trueString:
+		meetingID = zoomUser.Pmi
+	default:
+		meetingID, createMeetingErr = p.createMeetingWithoutPMI(user, zoomUser, req.ChannelID, topic)
 	}
 
-	meetingID := meeting.ID
-	if err = p.postMeeting(user, meetingID, req.ChannelID, "", req.Topic); err != nil {
-		http.Error(w, err.Error(), http.StatusInternalServerError)
+	if meetingID == -1 && createMeetingErr == nil {
 		return
 	}
 
-	p.trackMeetingStart(userID, telemetryStartSourceWebapp)
-
-	if r.URL.Query().Get("force") != "" {
-		p.trackMeetingForced(userID)
+	if meetingID >= 0 && createMeetingErr == nil {
+		if err = p.postMeeting(user, meetingID, req.ChannelID, "", topic); err == nil {
+			meetingURL := p.getMeetingURL(user, meetingID)
+			if _, err = w.Write([]byte(fmt.Sprintf(`{"meeting_url": "%s"}`, meetingURL))); err != nil {
+				p.API.LogWarn("failed to write response", "error", err.Error())
+			}
+			return
+		}
 	}
+	http.Error(w, err.Error(), http.StatusInternalServerError)
+}
 
-	meetingURL := p.getMeetingURL(user, meetingID)
-	_, err = w.Write([]byte(fmt.Sprintf(`{"meeting_url": "%s"}`, meetingURL)))
+func (p *Plugin) createMeetingWithoutPMI(user *model.User, zoomUser *zoom.User, channelID string, topic string) (int, error) {
```

Comment:
Check this everywhere
```suggestion
func (p *Plugin) createMeetingWithoutPMI(user *model.User, zoomUser *zoom.User, channelID, topic string) (int, error) {
```